src/convertToObject.js

```diff
@@ -7,6 +7,21 @@
  */
 function convertToObject(sourceString) {
   // write your code here
+  return sourceString.split(';').reduce((resultObj, segment) => {
+    const cleanSegment = segment.trim();
+
+    if (!cleanSegment) {
+      return resultObj;
+    }
+
+    const [key, data] = cleanSegment.split(':').map((part) => part.trim());
+
+    if (key && data) {
+      resultObj[key] = data;
+    }
+
+    return resultObj;
```

Comment:
Consider renaming 'resultObj' to a more descriptive name that reflects its purpose, such as 'styleObject' or 'cssProperties', to improve code readability.